cookbooks/aws-parallelcluster-config/recipes/fsx_mount.rb

```diff
@@ -38,32 +44,64 @@
   dns_name = if fsx_dns_name && !fsx_dns_name.empty?
                fsx_dns_name
              else
-               # Hardcoded DNSname only valid for filesystem created after Mar-1 2021
-               # For older filesystems, DNSname needs to be retrieved from FSx API
+               # Hardcoded DNS name only valid for lustre file systems created after Mar-1 2021
+               # For older file systems, DNS names need to be retrieved from FSx API
+               # For Ontap and OpenZFS file systems, the DNS name is always passed from CLI.
                "#{fsx_fs_id}.fsx.#{node['cluster']['region']}.amazonaws.com"
              end
+  case fsx_fs_type
+  when 'LUSTRE'
+    mount_name = fsx_mount_name_array[index]
+    mount_options = %w(defaults _netdev flock user_xattr noatime)
 
-  mount_name = fsx_mount_name_array[index]
-  mount_options = %w(defaults _netdev flock user_xattr noatime)
-
-  mount_options.concat(%w(noauto x-systemd.automount))
+    mount_options.concat(%w(noauto x-systemd.automount))
 
-  # Mount FSx over NFS
-  mount fsx_shared_dir do
-    device "#{dns_name}@tcp:/#{mount_name}"
-    fstype 'lustre'
-    dump 0
-    pass 0
-    options mount_options
-    action %i(mount enable)
-    retries 10
-    retry_delay 6
-  end
+    # Mount FSx over NFS
+    mount fsx_shared_dir do
+      device "#{dns_name}@tcp:/#{mount_name}"
+      fstype 'lustre'
+      dump 0
+      pass 0
+      options mount_options
+      action %i(mount enable)
+      retries 10
+      retry_delay 6
+    end
 
-  # Make sure permission is correct
-  directory fsx_shared_dir do
-    owner 'root'
-    group 'root'
-    mode '1777'
+    # Make sure permission is correct
+    directory fsx_shared_dir do
```

Comment:
The check we make on directory permissions is the same for every type of FSx. What about putting the check once in common for all after the mounting operations instead of repeating it?

Reply:
The check on permission is only done for Ontap and Lustre

Reply:
Two code sections with the same content are already a code duplication. What about writing the permission check block once within an unless block excluding OpenZFS and adding a comment beside it explaining why the check is skipped for OpenZFS?

Reply:
Done